apps/plea/tests/test_stages_data.py

```diff
@@ -577,6 +583,43 @@ def test_offences_not_shown_with_no_dx(self):
 
         self.assertFalse(getattr(stage.form, "case_data", False))
 
+    def test_offence_data_is_displayed_in_welsh(self):
+
+        activate("cy")
+
+        self.data["dx"] = True
+        self.data["plea"] = {}
+        self.data["case"]["urn"] = "06AA0000015"
+
+        stage = PleaStage(self.urls, self.data)
+        stage.load_forms({})
+        response = stage.render({})
+
+        offence = self.case.offences.all().first()
+
+        self.assertIn(offence.offence_short_title_welsh.encode("utf-8"), response.content)
+        self.assertIn(offence.offence_wording_welsh.encode("utf-8"), response.content)
+
+    def test_offence_data_falls_back_to_en_when_cy_data_is_not_available(self):
+
+        activate("cy")
+
+        offence = self.case.offences.all().first()
+
+        offence.offence_short_title_welsh = None
+        offence.offence_wording_welsh = None
+        offence.save()
+
+        self.data["dx"] = True
```

Comment:
These 6 lines look the same as in previous test; could they be refactored into a helper function?